cypress/e2e/awx/administration/wfApprovalsList.cy.ts

```diff
@@ -0,0 +1,595 @@
+  let userWFDeny: AwxUser;
+  let userWFCancel: AwxUser;
+  let inventory: Inventory;
+  let jobTemplate: JobTemplate;
+  let workflowJobTemplate: WorkflowJobTemplate;
+  let jobTemplateNode: WorkflowNode;
+  let approvalWFNode: WorkflowNode;
+  let jobName = '';
+
+  before(function () {
+    cy.awxLogin();
+  });
+
+  beforeEach(function () {
+    cy.createAwxOrganization().then((org) => {
+      organization = org;
+
+      cy.createAwxProject(
+        { organization: organization.id },
+        'https://github.com/ansible/test-playbooks'
+      ).then((proj) => {
+        project = proj;
+
+        cy.createAwxUser(organization).then((u) => {
+          user = u;
+        });
+        cy.createAwxUser(organization).then((u) => {
+          userWFApprove = u;
+        });
+        cy.createAwxUser(organization).then((u) => {
+          userWFDeny = u;
+        });
+        cy.createAwxUser(organization).then((u) => {
+          userWFCancel = u;
+        });
+        cy.createAwxInventory({ organization: organization.id })
+          .then((i) => {
+            inventory = i;
+          })
+          .then(() => {
+            cy.createAwxJobTemplate(
+              {
+                organization: organization.id,
+                project: project.id,
+                inventory: inventory.id,
+              },
+              'hello world.yml'
+            ).then((jt) => {
+              jobTemplate = jt;
+            });
+          });
+      });
+    });
+  });
+
+  afterEach(() => {
+    cy.deleteAwxJobTemplate(jobTemplate, { failOnStatusCode: false });
+    cy.deleteAwxInventory(inventory, { failOnStatusCode: false });
+    cy.deleteAwxProject(project, { failOnStatusCode: false });
+    cy.deleteAwxUser(user, { failOnStatusCode: false });
+    cy.deleteAwxUser(userWFApprove, { failOnStatusCode: false });
+    cy.deleteAwxUser(userWFCancel, { failOnStatusCode: false });
+    cy.deleteAwxUser(userWFDeny, { failOnStatusCode: false });
+  });
+
+  describe('Workflow Approvals - Approve, Deny, Delete', () => {
+    it('admin can approve and then delete a workflow approval from the list row item', () => {
+      cy.createAwxWorkflowJobTemplate({
+        name: 'E2E Workflow Approval-APPROVE-' + randomString(4),
+        organization: organization.id,
+        inventory: inventory.id,
+      }).then((wfjt) => {
+        workflowJobTemplate = wfjt;
+        cy.createAwxWorkflowVisualizerJobTemplateNode(workflowJobTemplate, jobTemplate).then(
+          (jtNode) => {
+            jobTemplateNode = jtNode;
+            cy.createAwxWorkflowVisualizerApprovalNode(workflowJobTemplate).then((appNode) => {
+              approvalWFNode = appNode;
+              cy.createWorkflowJTAlwaysNodeLink(approvalWFNode, jobTemplateNode);
+
+              cy.intercept(
+                'GET',
+                awxAPI`/workflow_job_templates/${workflowJobTemplate.id.toString()}/`
+              ).as('thisWfjt');
+              cy.visit(
```

Comment:
We should not be using `cy.visit` and we should not be hard coding url paths in the tests.
I'd like our tests to be uniform, use the custom commands cy.navigate and cy.clickTab, they both will handle this properly for us, thanks
I have updated tests using hardcoded URLs using cy.visit, specifically in WFVZr Job Output and other tests.

Reply:
if the action you are doing is getting to the details page, and then navigating to the details tab.

1. use cy.navigate of that nav item
2. then filter, search for the resource, click on it
3. go to the details page and do the assertions